src/sdf_xarray/__init__.py

```diff
@@ -250,6 +250,34 @@ def _process_grid_name(grid_name: str, transform_func) -> str:
             renamed_name = _rename_with_underscore(transformed_name)
             return renamed_name
 
+        def _process_latex_name(variable_name: str) -> str:
+            """Converts variable names to LaTeX format where possible
+            using the following rules:
+            - E -> E$_x$
+            - E -> E$_y$
+            - E -> E$_z$
+
+            This repeats for B, J and P. It only changes the variable
+            name if there are spaces around the affix (prefix + suffix)
+            or if there is no trailing space. This is to avoid changing variable
+            names that may contain these affixes as part of the variable name itself.
+            """
+            prefixes = ["E", "B", "J", "P"]
+            suffixes = ["x", "y", "z"]
+            for prefix in prefixes:
+                for suffix in suffixes:
+                    affix_spaces = f" {prefix}{suffix} "
+                    affix_no_trailing_space = f" {prefix}{suffix}"
+                    if affix_spaces in variable_name:
+                        return variable_name.replace(
+                            affix_spaces, f" {prefix}$_{suffix}$ "
```

Comment:
Generally all looks great, nice one!

This is entirely down to personal taste, but when I'm doing $\LaTeX$ stuff in Matplotlib I normally prefer to put the whole word in equation-mode:

```suggestion
                            affix_spaces, f" ${prefix}_{suffix}$ "
```

Is there any reason to prefer the method here where only the subscript is formatted, or is it just an aesthetics thing?

Reply:
While I do completely agree with moving the `$` to the start of the word I didn't do it here purely because I don't think we want the prefixes to be italicized. Normally when I write the electric field in the x component I write $\text{E}_x$ and not $E_x$ but sadly including the `\text{}` part in the string causes matplotlib to put everything after the formatted section on a new line which doesn't look greatl

Reply:
I do think this is better as $$E_x$$ rather than E $_x$. In the latter case the `E` won't be rendered as math upright but ordinary text, which is a little incongruous

Reply:
After a nights sleep I do agree with this change

Reply:
Fixed in https://github.com/PlasmaFAIR/sdf-xarray/pull/38/commits/14a4b654513d34613ca3e68ae4bb1ba95be9bb04